### DISCUSSION TITLE
testsuite: "complete-run.pl.in" file is not replaced by "complete-run.pl" script

### DISCUSSION THREAD

**Discussion by alissonmarcs** [i3/i3#6033] [Q&A] [answered]
Hello!

In order to run testsuite, I run these commands:

```shell
$ cd ~/i3/testcases
$ sudo apt-get install cpanminus
$ sudo cpanm .
$ cd ~/i3/AnyEvent-I3
$ sudo cpanm Module::Install
$ sudo cpanm .
```

I don't end up with `./complete-run.pl` script, so I run `./complete-run.pl.in` anyway, but get this error:

```shell
Can't locate SocketActivation.pm in @INC (you may need to install the SocketActivation module) (@INC entries checked: /home/alisson/repos/oss/i3/build/AnyEvent-I3/blib/lib @abs_top_builddir@ @abs_top_builddir@/testcases/lib @abs_top_srcdir@/testcases/lib @abs_top_builddir@/AnyEvent-I3/blib/lib /home/alisson/perl5/lib/perl5/5.38.2/x86_64-linux-thread-multi /home/alisson/perl5/lib/perl5/5.38.2 /home/alisson/perl5/lib/perl5/x86_64-linux-thread-multi /home/alisson/perl5/lib/perl5 /usr/lib/perl5/5.38/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.38/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.38/core_perl /usr/share/perl5/core_perl) at /home/alisson/perl5/lib/perl5/i3test/Util.pm line 17.
BEGIN failed--compilation aborted at /home/alisson/perl5/lib/perl5/i3test/Util.pm line 17.
Compilation failed in require at ./complete-run.pl.in line 22.
BEGIN failed--compilation aborted at ./complete-run.pl.in line 22.
```

I figure out that missing module is in `testcase/lib`, so I add it to `PERL5LIB` environment var and try again, but get other error:

```shell
$ export PERL5LIB="/home/alisson/repos/oss/i3/testcases/lib:$PERL5LIB"
$ ./complete-run.pl.in
Possible unintended interpolation of @abs_top_srcdir in string at ./complete-run.pl.in line 159.
Possible unintended interpolation of @abs_top_srcdir in string at ./complete-run.pl.in line 164.
Global symbol "@abs_top_srcdir" requires explicit package name (did you forget to declare "my @abs_top_srcdir"?) at ./complete-run.pl.in line 159.
Global symbol "@abs_top_srcdir" requires explicit package name (did you forget to declare "my @abs_top_srcdir"?) at ./complete-run.pl.in line 164.
BEGIN not safe after errors--compilation aborted at ./complete-run.pl.in line 428.
```

I don't know how to proceed. I expected that after run the commands to install dependencies, I would have the `./complete-run.pl` script.

Some points:

- First time I run the commands to install dependencies, I run without `sudo`. This is because I did not pay attention when reading documentation, but after that I run with `sudo`.
- My Perl environment vars are as follows:

```shell
$ env | grep PERL
PERL5LIB=/home/alisson/repos/oss/i3/testcases/lib:/home/alisson/perl5/lib/perl5
PERL_MB_OPT=--install_base "/home/alisson/perl5"
PERL_MM_OPT=INSTALL_BASE=/home/alisson/perl5
PERL_LOCAL_LIB_ROOT=/home/alisson/perl5
```
- I tried with `cpan` but not success.
- When I try `make check` in `build` folder:

```
$ make check
make: *** No rule to make target 'check'.  Stop.
```

I tried to follow documentation links to `i3test` and `i3test::Test` in [testsuite](https://i3wm.org/docs/testsuite.html#_relevant_documentationl), but are broken. I also tried [metacpan](https://metacpan.org/) but not found.

Any help are welcome =)

**Comment by orestisfl** [comment]
`ninja test` is the command you should be running, ignore anything with `make _`, it is outdated. I also recommend to set up a local perl library instead of running `sudo cpanm`, see https://michael.stapelberg.ch/cpan/

**Reply by orestisfl** [reply, marked as answer]
To be precise, if you have all the perl dependencies in place this should be enough to run the tests from a clean clone of the i3 repo:
```
mkdir -p build
meson setup build
cd build
ninja test
```

**Reply by alissonmarcs** [reply]
@orestisfl thank you very much
